hail/python/hail/fs/router_fs.py

```diff
@@ -230,10 +238,9 @@ async def size():
 
     def ls(self, path: str, _max_simultaneous_files: int = 50) -> List[Dict]:
         async def _ls():
-            async with OnlineBoundedGather2(asyncio.Semaphore(_max_simultaneous_files)) as pool:
-                tasks = [pool.call(self._fle_to_dict, fle)
-                         async for fle in await self.afs.listfiles(path)]
-                return [await t for t in tasks]
+            return await bounded_gather(
+                *[functools.partial(self._fle_to_dict, fle)
```

Comment:
Do we not care how many simultaneous queries we do?

Reply:
Ah, yes, I dropped the parallelism argument. I restored it